WordPress/gutenberg
# Columns: Add border support

Related: https://github.com/WordPress/gutenberg/issues/21889

## Description
- Opts into border block support for the Columns block
- No changes to the inner Column blocks have been made regarding border support. They will be addressed in the near future when support for individual border configurations is available.

<details>
<summary>Previous PR Description (included block style approach to dividing borders for inner columns)</summary>

## Description
- Adds control of Columns block border
- Adds control of inner Column block right hand borders
- Adds "divided" block style where only right border is shown or only bottom border on mobile

##### Notes
- Until https://github.com/WordPress/gutenberg/issues/31337 is resolved the UI for this won't be enabled. (See https://github.com/WordPress/gutenberg/pull/31585 for WIP PR)
- Inner column borders are controlled individually. If we want a single set of controls for them all, an ad hoc solution needs to be implemented at the Columns block level rather than using block supports
- Positioning of the borders isn't ideal however will be addressed separately when margin & padding support is added to columns
- The naming of the "Divided" block style could use improving. Open to suggestions for that.

## Block patterns that may use this feature
- https://github.com/WordPress/gutenberg/issues/21889#issuecomment-673829155
- https://github.com/WordPress/gutenberg/issues/21889#issuecomment-697004993

## Screenshots 
<img src="https://user-images.githubusercontent.com/60436221/119456534-1b828680-bd7e-11eb-91f4-51bca77dc53c.gif" width="600"/>
</details>

## How has this been tested?
Manually.

#### Setup
- Ensure your current theme has the border support UI enabled or has `appearanceTools: true` under settings.

#### Test Instructions

1. Create a post, add a columns block, some inner columns and give them some content
2. Select the outer columns block and confirm border controls in the sidebar work
3. Select an inner column block and note the absence of border control (for the time being)
4. Apply a border to the main columns block, save the post and confirm correct rendering on frontend
5. Back in the block editor, remove the border and save
6. Switch to the site editor and via the Global Styles sidebar navigate to Blocks > Columns. Then apply a generic border.
7. Confirm the global border style for Columns blocks renders in the site editor, block editor, and frontend.

## Screenshots 
![ColumnBordersDemo](https://user-images.githubusercontent.com/60436221/153983747-cb02dfe7-051a-4aed-b1ea-296868ef24cc.gif)

## Types of changes
New feature.

## Checklist:
